rapid_response_xblock/block.py

```diff
@@ -153,31 +155,45 @@ def responses(self, request=None, suffix=None):  # pylint: disable=unused-argume
         """
         Returns student responses for rapid-response-enabled block
         """
-        status = RapidResponseBlockStatus.objects.filter(
-            problem_usage_key=self.wrapped_block_usage_key,
-            course_key=self.course_key
-        ).first()
-        is_open = False if not status else status.open
-        response_data = RapidResponseSubmission.objects.filter(
+        runs = RapidResponseRun.objects.filter(
             problem_usage_key=self.wrapped_block_usage_key,
             course_key=self.course_key,
-        ).values('answer_id').annotate(count=Count('answer_id'))
-        response_counts = {item['answer_id']: item['count'] for item in response_data}
+        ).order_by('created')
+        is_open = runs.filter(open=True).exists()
```

Comment:
It _should_ only be possible for the most recent `RapidResponseRun` to have `open=True`, right? If not, I think that presents some problems with the overall logic. If so, we should just check the first element of the queryset for `open=True` (assuming descending order) rather than executing another query. Also, I think it's better not to set this variable and break up the logic. IMO it's more readable just to evaluate when we're building the Response json_body

Reply:
Yes it's only the most recent one, I'll update the expression